block-producer/src/block_builder/prover/block_witness.rs

```diff
@@ -189,11 +130,12 @@ impl BlockWitness {
         }
 
         Self::validate_account_states(block_inputs, batches)?;
+        Self::validate_nullifiers(block_inputs, batches)?;
```

Comment:
We should probably remove the `TODO` on line 125 above.

Reply:
done